outreach/views/order.py

```diff
@@ -0,0 +1,109 @@
+            items_json[unicode(line_item.item_id)]['final_amount'] += line_item.base_amount
+        else:
+            items_json[unicode(line_item.item_id)]['final_amount'] = None
+        items_json[unicode(line_item.item_id)]['quantity'] += 1
+        items_json[unicode(line_item.item_id)]['quantity_available'] = item.get_available_quantity()
+    return items_json
+
+
+@app.route('/order/kharcha', methods=['OPTIONS', 'POST'])
+@xhr_only
+@cors
+def kharcha():
+    """
+    Accepts JSON containing an array of line_items, with the quantity and item_id set for each line_item.
+
+    Returns JSON of line items in the format:
+    {item_id: {'quantity': Y, 'final_amount': Z}}
+    """
+    if not request.json or not request.json.get('line_items'):
+        return make_response(jsonify(message='Missing line items'), 400)
+    line_item_forms = LineItemForm.process_list(request.json.get('line_items'))
+    if not line_item_forms:
+        return make_response(jsonify(message='Invalid line items'), 400)
+
+    # Make line item splits and compute amounts
+    line_items = LineItem.calculate([{'item_id': li_form.data.get('item_id')}
+        for li_form in line_item_forms
+            for x in range(li_form.data.get('quantity'))])
+    items_json = jsonify_line_items(line_items)
+    order_final_amount = sum([values['final_amount'] for values in items_json.values() if values['final_amount'] is not None])
+    return jsonify(line_items=items_json, order={'final_amount': order_final_amount})
+
+
+@app.route('/ic/<item_collection>/inquiry',
+           methods=['OPTIONS', 'POST'])
+@load_models(
+    (ItemCollection, {'id': 'item_collection'}, 'item_collection')
+    )
+@xhr_only
+@cors
+def inquiry(item_collection):
+    buyer_form = BuyerForm.from_json(request.json.get('buyer'))
+    # See comment in BuyerForm about CSRF
+    buyer_form.csrf_enabled = False
+    if not buyer_form.validate():
+        return make_response(jsonify(message='Invalid buyer details'), 400)
+
+    user = User.query.filter_by(email=buyer_form.email.data).first()
+
+    order = Order(user=user,
+        organization=item_collection.organization,
+        item_collection=item_collection,
+        buyer_email=buyer_form.email.data,
+        buyer_fullname=buyer_form.fullname.data,
+        buyer_company=buyer_form.company.data,
+        buyer_phone=buyer_form.phone.data)
+
+    order.make_inquiry()
+
+    line_item_forms = LineItemForm.process_list(request.json.get('line_items'))
+    if line_item_forms:
+        line_item_tups = LineItem.calculate([{'item_id': li_form.data.get('item_id')}
+            for li_form in line_item_forms
+                for x in range(li_form.data.get('quantity'))])
+        for idx, line_item_tup in enumerate(line_item_tups):
+            item = SaleItem.query.get(line_item_tup.item_id)
+            line_item = LineItem(order=order, sale_item=item,
+                seq=idx+1,
```

Comment:
Don't do this. [`enumerate` takes a start parameter](https://docs.python.org/2/library/functions.html#enumerate).